include/swift/AST/Module.h

```diff
@@ -209,7 +208,23 @@ class ModuleDecl : public DeclContext, public TypeDecl {
 
 public:
   typedef ArrayRef<Located<Identifier>> AccessPathTy;
-  typedef std::pair<ModuleDecl::AccessPathTy, ModuleDecl*> ImportedModule;
+  struct ImportedModule {
+    ModuleDecl::AccessPathTy accessPath;
+    ModuleDecl *importedModule;
+  
+    /// Will not accept a null moduleDecl pointer.
```

Comment:
I think it would be better to document this at the level of the field instead of the constructor because this is an invariant of the type itself, not of the specific constructor. Say something like

```
/// Invariant: The pointer is not null.
ModuleDecl *importedModule;
```

What do you think?